iquip/apps/scheduler.py

```diff
@@ -30,6 +34,40 @@ def _dismiss_items(layout: Optional[QLayout] = None):
                 _dismiss_items(item.layout())
 
 
+def _run_thread_with_worker(worker: QObject):
+    """Runs another thread with given worker.
+
+    Args:
+        worker: The worker that must be run through another thread. It must have:
+          - worker.run: the main function that has to be run.
+          - worker.done: the signal that is emitted when the work is done.
+    """
+    thread = QThread()
+    worker = SchedulerPostWorker("delete")
+    worker.moveToThread(thread)
+    thread.started.connect(worker.run)
+    worker.done.connect(thread.quit)
+    worker.done.connect(worker.deleteLater)
+    thread.finished.connect(thread.deleteLater)
+    thread.start()
+
+
+class ExperimentListView(QListView):
+    """Customized QListView class to detect right-click input.
+    
+    Signals:
+        fetched_event(QMouseEvent): The information of the click input is sent.
+    """
+    fetched_event = pyqtSignal(QMouseEvent)
```

Comment:
Please use the camelcase: `fetchedEvent`.
BTW what is the meaning of its name? For me something like `rightButtonPressed` would be more natural.